In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline

In [2]:
psp = pd.read_csv('price_sales_promotion_cleaned.csv')
psp.head()

Agency     SKU  YearMonth        Price        Sales  Promotions
0  Agency_01  SKU_01     201301  1141.500000  1033.432731  108.067269
1  Agency_01  SKU_01     201302  1141.500000  1065.417195   76.082805
2  Agency_01  SKU_01     201303  1179.345820  1101.133633   78.212187
3  Agency_01  SKU_01     201304  1226.687500  1138.283357   88.404143
4  Agency_01  SKU_01     201305  1230.331104  1148.969634   81.361470

There are 18437 entries.  
There are 58 agencies(Agency_01 to Agency_60, except 06 & 14).  
There are 25 SKUs(SKU_01-SKU_34).  
5 Years worth of data from 201301 to 201712 is present.  
Price ranges from 984 to 4925 with mean of 1651.  
Sales ranges from 654 to 4925 with mean of 1449.  
Promotions has a range from 0 to 1964 with mean of 202.37.  

In [3]:
hv = pd.read_csv('historical_volume_cleaned.csv')
hv.head()

Agency     SKU  YearMonth    Volume
0  Agency_22  SKU_01     201301   52.2720
1  Agency_22  SKU_02     201301  110.7000
2  Agency_48  SKU_07     201301   28.3200
3  Agency_22  SKU_05     201301  238.5387
4  Agency_31  SKU_01     201301  126.3600

There are 18449 entries.  
The yearmonth, agency and sku count are similar to psp file.  
Volume ranges from 0.03 to 22526 with a mean of 1698.  

In [4]:
weather = pd.read_csv('weather.csv')
weather.head()

YearMonth     Agency  Avg_Max_Temp
0     201301  Agency_01        17.072
1     201302  Agency_01        19.984
2     201303  Agency_01        24.600
3     201304  Agency_01        27.532
4     201305  Agency_01        29.396

Weather data is provided for all agencies(58+2) till Dec 2017. The data is not provided for the next month. The data looks ok.  
Ave_Max_Temp takes values from 16 to 45 with a mean of 28. 

In [5]:
iss = pd.read_csv('industry_soda_sales.csv')
iss.head()

YearMonth  Soda_Volume
0     201301    718394219
1     201302    753938444
2     201303    892192092
3     201304    838099501
4     201305    864420003

The Soda sale data covers the past 5 years.  Values range from 696 million to a billion.

In [6]:
ec = pd.read_csv('event_calendar.csv')
ec.head()

YearMonth  Easter Day  Good Friday  New Year  Christmas  Labor Day  \
0     201301           0            0         1          0          0   
1     201302           0            0         0          0          0   
2     201303           1            1         0          0          0   
3     201304           0            0         0          0          0   
4     201305           0            0         0          0          1   

   Independence Day  Revolution Day Memorial  Regional Games   \
0                 0                        0                0   
1                 0                        0                0   
2                 0                        0                0   
3                 0                        0                0   
4                 0                        0                0   

   FIFA U-17 World Cup  Football Gold Cup  Beer Capital  Music Fest  
0                    0                  0             0           0  
1                    0                  0             0           0  
2                    0                  0             0           1  
3                    0                  0             0           0  
4                    0                  0             0           0

Has events for time periods including Jan-18. Two football columns are empty

In [7]:
iv = pd.read_csv('industry_volume.csv')
iv.head()

YearMonth  Industry_Volume
0     201301        492612703
1     201302        431937346
2     201303        509281531
3     201304        532390389
4     201305        551755254

Industry volume covers past 5 years. Volume ranges from 400m to 670m with mean of 540m

In [8]:
dem = pd.read_csv('demographics.csv')
dem.head()

Agency  Avg_Population_2017  Avg_Yearly_Household_Income_2017
0  Agency_05              3044268                            182944
1  Agency_02              3137874                            240809
2  Agency_55              2025793                            238553
3  Agency_53              1729177                            111048
4  Agency_50              2057678                            144594

Has demographics data for all 60 agencies. The agencies are in different areas. The average population for 2017 is between 12k and 3m with mean of around 1m. The average yearly household income ranges from 90k to 247k with 148k as the mean.

In [9]:
vf = pd.read_csv('test/volume_forecast.csv')
vf.head()

Agency     SKU  Volume
0  Agency_23  SKU_06     NaN
1  Agency_34  SKU_20     NaN
2  Agency_24  SKU_02     NaN
3  Agency_40  SKU_06     NaN
4  Agency_42  SKU_32     NaN

Volume forecasting is for the 58 agency and 25 SKUs